codegenerator/cli/src/config_parsing/system_config.rs

```diff
@@ -791,10 +798,19 @@ impl Contract {
     }
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub enum EventPayload {
+    Params(Vec<EventParam>),
+    Data(RescriptTypeIdent),
```

Comment:
I'm pretty keen to use the same way of generating types for EVM in future. It's nice because it easily supports nested record types which we currently just inline as tuples.

Reply:
Sounds good, the only problem I see is building a hyperSyncConverter function

Reply:
Hmmm, yeah, but we should be able to lookup nested types and codegen a converter without too much trouble.

Reply:
What about the `is_indexed`?

Reply:
> Should work, but what about the `is_indexed`?

Well, we should generate the converter from the EVM event, not the rescript type. If you add an index to a nested struct/tuple/array, the actual data doesn't get stored on chain. Just a keccak hash so we don't really have to handle params that are a nested index structure.